updated docs for date picker to new format

Zadielerick opened the pull request:
Need feedback. Description for value and value Link. ShowYearSelector seems to be misnamed from disableYearSelection that is passed down to Calendar. Thoughts? Also on the description of the Date Picker.

oliviertassinari commented on docs/src/app/components/DatePicker/ExampleDatePickers.jsx:
```diff
@@ -0,0 +1,121 @@
+import Toggle from 'material-ui/lib/toggle';
+
+
+const DatePickerExamples = React.createClass({
```
The name is not consistent with `docs/src/app/components/DatePicker/ExampleDatePickers.jsx`
Can we call this class `DatePickerExampleSimple` and the file `ExampleSimple.jsx`?

oliviertassinari replied:
Can we use the ES6 class too?

Zadielerick replied:
@oliviertassinari Is there somewhere else with an example of ES6 class in material-ui?

oliviertassinari replied:
Have a look at https://github.com/callemall/material-ui/blob/master/docs/src/app/components/Snackbar/ExampleTwice.jsx.

oliviertassinari commented on docs/src/app/components/DatePicker/ExampleDatePickers.jsx:
```diff
@@ -0,0 +1,121 @@
+  },
+
+  render() {
+    let optionsStyle = {
```
No need  to allocate the memory for the style at each render, can we move it outside?

oliviertassinari commented on docs/src/app/components/DatePicker/README.md:
```diff
@@ -0,0 +1,3 @@
+## Date Picker
+
```
Can you add the description to https://www.google.com/design/spec/components/pickers.html#pickers-date-pickers?

oliviertassinari commented on src/date-picker/date-picker.jsx:
```diff
@@ -30,31 +30,128 @@ const DatePicker = React.createClass({
 
   propTypes: {
+    /**
+      * Constructor for time formatting.
```
We have one extra space.
```

  /**
   * Description...
   */
```

oliviertassinari commented on src/date-picker/date-picker.jsx:
```diff
@@ -30,31 +30,128 @@ const DatePicker = React.createClass({
+      * `inline` displays the DatePicker below the input field (similar to auto complete)
+      */
     container: React.PropTypes.oneOf(['dialog', 'inline']),
+
```
@subjectix Should we add the className property too?

oliviertassinari commented on docs/src/app/components/DatePicker/ExampleSimple.jsx:
```diff
@@ -0,0 +1,123 @@
+import Toggle from 'material-ui/lib/toggle';
+
+const optionsStyle = {
+  width: '300px',
```
That's better here :+1:.
Since it's for documentation, I would say that
```js
  width: 300,
```
is better

oliviertassinari commented:
@subjectix Where are you with https://github.com/callemall/material-ui/pull/2585?
Can we change the doc? Otherwise, can you split your PR with the doc bit if it's finished so that we can keep merging PR?

subjectix commented:
@oliviertassinari I have to do it all at once, that change is rooted very deep into implementation. I'll fix things up tonight. Hopefully it will all workout this time :grin:

oliviertassinari commented:
@subjectix Alright, good luke :+1:

subjectix commented:
@oliviertassinari You can merge this if you'd like I'm gonna rebase

subjectix commented:
@oliviertassinari We should merge this fast, so I can rebase #2628 onto this.

oliviertassinari commented:
@Zadielerick The rebase should be straighforward since the conflict commit is a simple `git mv`

oliviertassinari commented on src/date-picker/date-picker.jsx:
```diff
@@ -30,31 +30,128 @@ const DatePicker = React.createClass({
     shouldDisableDate: React.PropTypes.func,
+
+    /**
+     *
```
Missing documentation?

Zadielerick replied:
@oliviertassinari Yeah, it looks like the prop name changes to disableYearSelector as it is passed down to Calendar. Not sure though, wanted to check with yall before changing this

oliviertassinari replied:
Yeah, sorry, I have no idea about it. I should look at the source code.

oliviertassinari replied:
Can you point me the line?

Zadielerick replied:
@oliviertassinari If you follow the prop, it is passed down to date picker dialog. [Here](https://github.com/callemall/material-ui/blob/master/src/date-picker/date-picker-dialog.jsx#L183) it is passed down to calendar, but calendar has no such prop, but it has [disableYearSelection](https://github.com/callemall/material-ui/blob/master/src/date-picker/calendar.jsx#L41) which isn't called anywhere else. Thought it might be misnamed.

oliviertassinari replied:
@Zadielerick Actually, I feel like it's an dead property that we should remove.

oliviertassinari replied:
I see now, thanks for the explaination! :flushed:

Zadielerick replied:
Should I rename?

oliviertassinari replied:
That's definitely broken.I think that the best thing to do is to use `disableYearSelection` and remove `showYearSelector`.

oliviertassinari commented on src/date-picker/date-picker.jsx:
```diff
@@ -30,31 +30,128 @@ const DatePicker = React.createClass({
     value: React.PropTypes.any,
+
+    /**
+     * Creates a ValueLink with the value of date picker
```
I would say, that the description is a sentence, hence it's missing a dot at the end :grin:.

oliviertassinari commented on src/date-picker/date-picker.jsx:
```diff
@@ -30,31 +30,128 @@ const DatePicker = React.createClass({
     textFieldStyle: React.PropTypes.object,
+
+    /**
+     * Sets the date for the Date Picker programmatically
```
`.`

oliviertassinari commented on src/date-picker/date-picker.jsx:
```diff
@@ -30,31 +30,128 @@ const DatePicker = React.createClass({
     onShow: React.PropTypes.func,
+
+    /**
+     *
```
Missing description? Maybe it's not needed.

oliviertassinari commented on src/date-picker/date-picker.jsx:
```diff
@@ -30,31 +30,128 @@ const DatePicker = React.createClass({
     formatDate: React.PropTypes.func,
+
+    /**
+     *
```
?

Zadielerick replied:
@oliviertassinari I don't see it being used in Date Picker Dialog. Maybe its been removed?

oliviertassinari replied:
Looks like it's a dead property. This component is kind of a mess :grin:

Zadielerick replied:
:+1:

oliviertassinari commented on src/date-picker/date-picker.jsx:
```diff
@@ -30,31 +30,128 @@ const DatePicker = React.createClass({
+    /**
+     * Used to control how the DatePicker will be displayed when a user tries to set a date.
+     * `dialog` (default) displays the DatePicker as a dialog with a modal.
+     * `inline` displays the DatePicker below the input field (similar to auto complete)
```
`.`

Zadielerick commented:
@oliviertassinari Actually, there seems to be a bug with the toggles on the date picker component. Makes the datepicker become uncloseable, and with my pull request toggles dont do anything... Don't know whether thats good or bad...

oliviertassinari commented on docs/src/app/components/pages/components/DatePicker/ExampleSimple.jsx:
```diff
@@ -0,0 +1,123 @@
+    this.setState(state);
+  }
+
+  _handleChange = (nill, date) => {
```
`nill` -> `event`?

oliviertassinari commented:
> Makes the datepicker become uncloseable

How can I reproduce this?

oliviertassinari commented on docs/src/app/components/pages/components/DatePicker/ExampleSimple.jsx:
```diff
@@ -0,0 +1,123 @@
+    });
+  }
+
+  _handleToggle = (e, toggled) => {
```
I would say that `event` is more explicit than `e`.

Zadielerick replied:
:+1:

Zadielerick commented:
To reproduce the bug:
1) Go to date picker component on docs site and toggle the disable year selection toggle
2) Select the Inline Date Picker, click on the Year Selection at the top and change the Year
3) Date Picker Dialog will not close after that

oliviertassinari commented:
The behavior seems to be the same on the master branch than on your PR or on the material doc website.

Zadielerick commented:
Yeah, I think it has to do with that disableYearSelection prop. Since I changed it on my local right now, the behavior went away, but now it doesnt do anything.

subjectix commented:
@Zadielerick Please only rebase when your PR is ready to be merged. rebasing after each change removes the commits and makes reviewing recent changes harder. :grin:

subjectix commented:
@oliviertassinari I think the ExampleSimple is too big too be Simple :laughing: Do you think it's better to break it down into smaller examples that serve to demonstrate a single point? I mean one DatePicker component per example would keep it cleaner.

oliviertassinari commented:
Yeah, two or three examples would be better. We could have
- a simple
- an inline one
- a last one with the toggles

Zadielerick commented:
@oliviertassinari So, I fixed the bug and added 3 seperate examples. However, we still do not include an example with a Controlled Date Picker or a Date Picker using an International date format. Should I also add those as examples?

oliviertassinari commented:
> So, I fixed the bug and added 3 seperate examples.

Yeah :tada:.

> However, we still do not include an example with a Controlled Date Picker or a Date Picker using an International date format.

Is the controlled Date Picker feature working or in the existing examples? If not let's see later.
If think that we should have the international examples, as before :+1:.

Zadielerick commented:
Yes, controlled is working.

oliviertassinari commented:
> Yes, controlled is working, but is not used in existing examples.

Well, as you want.

Zadielerick commented:
ok, Ill incorporate it in one of the existing examples

oliviertassinari commented on docs/src/app/components/pages/components/DatePicker/Page.jsx:
```diff
@@ -0,0 +1,47 @@
+  require('intl/locale-data/jsonp/fr');
+}
+
+export default class DatePickerPage extends React.Component {
```
What about using a stateless functional component?

Zadielerick replied:
@oliviertassinari So like <p><code>const DatePickerPage = () => </code></p>

oliviertassinari replied:
Yes,
```jsx
const DatePickerPage = () => (
  <div />
);
```

Zadielerick replied:
:+1:

oliviertassinari commented on src/date-picker/date-picker.jsx:
```diff
@@ -30,31 +30,124 @@ const DatePicker = React.createClass({
+
+    /**
+     * This function is called to format the date to display in the input box.
+     * By default, date objects are formatted to M/D/YYYY.
```
`to MM/DD/YYYY` instead?

Zadielerick replied:
:+1:

oliviertassinari commented on src/date-picker/date-picker.jsx:
```diff
@@ -30,31 +30,124 @@ const DatePicker = React.createClass({
+     * false the day is disabled otherwise it is displayed normally.
+     */
     shouldDisableDate: React.PropTypes.func,
-    showYearSelector: React.PropTypes.bool,
```
What about using `deprecated` like here https://github.com/callemall/material-ui/blob/master/src/left-nav.jsx#L69.
to warn users that they should use `disableYearSelection` instead?
Since, it wasn't working before, I'm not sure.

Zadielerick replied:
Yeah, I will add that in case someone copied code and used it.

Zadielerick replied:
:+1:

oliviertassinari commented on docs/src/app/components/pages/components/DatePicker/ExampleInline.jsx:
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import DatePicker from 'material-ui/lib/date-picker/date-picker';
+
+export default class DatePickerExampleSimple extends React.Component {
```
Stateless functional component here too?

oliviertassinari replied:
`DatePickerExampleInline`

Zadielerick replied:
:+1:

oliviertassinari commented on docs/src/app/components/pages/components/DatePicker/ExampleInternational.jsx:
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import DatePicker from 'material-ui/lib/date-picker/date-picker';
+
+export default class DatePickerExampleSimple extends React.Component {
```
Stateless functional component?

Zadielerick replied:
:+1:

oliviertassinari commented on docs/src/app/components/pages/components/DatePicker/ExampleSimple.jsx:
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import DatePicker from 'material-ui/lib/date-picker/date-picker';
+
+export default class DatePickerExampleSimple extends React.Component {
```
Stateless functional component?

Zadielerick replied:
:+1:

oliviertassinari commented on docs/src/app/components/pages/components/DatePicker/ExampleInline.jsx:
```diff
@@ -0,0 +1,37 @@
+    super(props);
+
+    this.state = {
+      controlledDate: new Date('2015/07/15'),
```
Can we start with a null value? I which we can display the `Controlled Date Input` hint text.

Zadielerick replied:
:+1:

oliviertassinari replied:
The systematic :+1:, is flooding my mailbox. Could you stop :grin:?

Zadielerick replied:
Sorry :(

oliviertassinari commented on docs/src/app/components/pages/components/DatePicker/ExampleToggle.jsx:
```diff
@@ -0,0 +1,92 @@
+  margin: '0 auto',
+};
+
+export default class DatePickerExampleSimple extends React.Component {
```
`DatePickerExampleToggle`

Zadielerick replied:
:+1:

oliviertassinari commented on docs/src/app/components/pages/components/DatePicker/ExampleToggle.jsx:
```diff
@@ -0,0 +1,92 @@
+      minDate: minDate,
+      maxDate: maxDate,
+      autoOk: false,
+      controlledDate: new Date('2015/07/15'),
```
Not used.

Zadielerick replied:
:+1:

Zadielerick committed:
Temp

oliviertassinari commented on docs/src/app/components/pages/components/DatePicker/ExampleToggle.jsx:
```diff
@@ -0,0 +1,91 @@
+  }
+
+  _handleToggle = (event, toggled) => {
+    let state = {};
```
`const`

oliviertassinari commented on docs/src/app/components/pages/components/DatePicker/ExampleInternational.jsx:
```diff
@@ -0,0 +1,14 @@
+import DatePicker from 'material-ui/lib/date-picker/date-picker';
+
+const DatePickerExampleInternational = () => (
+      <DatePicker
```
indent

oliviertassinari commented on docs/src/app/components/pages/components/DatePicker/Page.jsx:
```diff
@@ -0,0 +1,41 @@
+}
+
+const DatePickerPage = () => (
+    <div>
```
indent

Zadielerick committed:
Updated Docs for Date Picker

alitaheri commented on src/date-picker/date-picker.jsx:
```diff
@@ -30,31 +31,129 @@ const DatePicker = React.createClass({
+    /**
+     * Creates a ValueLink with the value of date picker.
+     */
     valueLink: React.PropTypes.object,
```
@oliviertassinari Should we start deprecating this all over the place?

oliviertassinari replied:
I was wondering. I don't think that we should do it in this PR.

alitaheri replied:
you beat me to it :laughing:

alitaheri commented:
@Zadielerick Such fine job :+1: :+1: Thanks a lot :grin:

@oliviertassinari Lets merge this for now. we should make a big PR deprecating valuLinks

Zadielerick commented:
@alitaheri @oliviertassinari  Thanks! And thanks for checking my stuff, learning a lot from yall :)

oliviertassinari commented on src/date-picker/date-picker.jsx:
```diff
@@ -30,31 +31,129 @@ const DatePicker = React.createClass({
+    /**
+     *  Enables the year selection in the date picker.
+     */
+    showYearSelector: deprecated(React.PropTypes.element,
```
bool?

alitaheri commented:
While you're at it. can that controlled DatePicker be somewhere else (it's own example would be good too) because it's like we have two perfectly related inline pickers and then a dialog picker. it feels inconsistent. Sorry my last one I swear :grin: :grin:

Zadielerick commented:
Yeah, I wanted to incorporate the controlled date picker into an existing example or else we have 5 examples. Should I put it in another existing or make a new example for it?

alitaheri commented:
@Zadielerick I don't think having a lot of examples is a bad thing. In fact I would like a library with a lot of examples, it will surely have something for any of my needs. I'd like to hear @oliviertassinari's thoughts on this one.

Zadielerick committed:
Updated Docs for Date Picker

alitaheri commented:
I'm all green :+1: :+1: 

@oliviertassinari merge if you're good too :grin:

alitaheri commented:
Thank you @Zadielerick, This was some hard work :+1: :+1: :+1: :tada: :balloon:

Zadielerick commented:
:) No problem!

oliviertassinari commented:
@Zadielerick Thanks :smile:.!
For the next commits, I suggest you to follow this rule `Use the imperative mood in the subject line` http://chris.beams.io/posts/git-commit/.

Zadielerick commented:
@oliviertassinari I see, thank you!
